hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiveLog.java

```diff
@@ -253,4 +256,19 @@ private IndexedRecord convertToAvroRecord(HoodieTimeline commitTimeline,
             com.uber.hoodie.avro.model.HoodieCommitMetadata.class);
     return avroMetaData;
   }
+
+  /**
+   * Check and delete in-flight instants.
+   */
+  private void deleteInflightInstants() {
```

Comment:
ideally, we should not have the need the delete an inflight instants. An inflight instant for clean will remain if it failed. If we want to clean up older remnants of such failures, we should also check if the inflight is older than a successful future clean.

Reply:
To simplify this PR, we can just keep those inflight instants, basically removing the `deleteInflightInstants` altogether given that we shouldn't have much of those empty in-flights.

@vinothchandar thoughts?

Reply:
Similar thoughts.. Deleting all in-flight clean actions from archival does not seem like the right thing to do here.  What code is currently responsible for cleaning up a in-flight clean or rollback.  If its not clear, then thats where I want to start with.. 

also can you paste the entire stack trace with the exception and give me more context into when this is happening?

Reply:
More complete stack trace:
com.uber.hoodie.exception.HoodieCommitException: Failed to archive commits
```
at com.uber.hoodie.io.HoodieCommitArchiveLog.archive(HoodieCommitArchiveLog.java:201)                           
at com.uber.hoodie.io.HoodieCommitArchiveLog.archiveIfRequired(HoodieCommitArchiveLog.java:110)                 
at com.uber.hoodie.HoodieWriteClient.commit(HoodieWriteClient.java:415)                    
...                     
```
Caused by: java.io.IOException: Not an Avro data file
```
at org.apache.avro.file.DataFileReader.openReader(DataFileReader.java:50)                                       
at com.uber.hoodie.common.util.AvroUtils.deserializeAvroMetadata(AvroUtils.java:189)                            
at com.uber.hoodie.io.HoodieCommitArchiveLog.convertToAvroRecord(HoodieCommitArchiveLog.java:214)               
at com.uber.hoodie.io.HoodieCommitArchiveLog.archive(HoodieCommitArchiveLog.java:196)                           
... 13 more
```